sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncItemResponse.java

```diff
@@ -4,40 +4,41 @@
 
 import com.azure.cosmos.implementation.Document;
 import com.azure.cosmos.implementation.ResourceResponse;
+import com.azure.cosmos.implementation.Utils;
 import org.apache.commons.lang3.StringUtils;
 
-public class CosmosAsyncItemResponse extends CosmosResponse<CosmosItemProperties> {
-    private final CosmosAsyncItem itemClient;
-
-    CosmosAsyncItemResponse(ResourceResponse<Document> response, PartitionKey partitionKey,
-                            CosmosAsyncContainer container) {
+public class CosmosAsyncItemResponse<T> extends CosmosResponse<CosmosItemProperties> {
+    private final Class<T> itemClassType;
+    private final String responseBodyString;
+    
+    CosmosAsyncItemResponse(ResourceResponse<Document> response, Class<T> klass) {
         super(response);
-        String bodyAsString = response.getBodyAsString();
-        if (StringUtils.isEmpty(bodyAsString)) {
+        this.itemClassType = klass;
+        responseBodyString = response.getBodyAsString();
+        if (StringUtils.isEmpty(responseBodyString)){
             super.setProperties(null);
-            itemClient = null;
         } else {
-            CosmosItemProperties props = new CosmosItemProperties(bodyAsString);
+            CosmosItemProperties props = new CosmosItemProperties(responseBodyString);
```

Comment:
If the public API is `T getResource()` the instantiation of `CosmosItemProperties` will do double deserialization. We should remember to remove this altogether and solely rely on `T getResource()`

Reply:
Yes, next PR should be for removing CosmosItemProperties as a public type.